frontend/src/metabase/redux/uploads.ts

```diff
@@ -70,18 +70,20 @@ export const uploadFile = createThunkAction(
         },
       );
 
-      if (!response) {
-        return;
-      }
+      if (response) {
```

Comment:
Is having no response a valid case? Should we dispatch `uploadError` when there is no response?

Reply:
if there's a 200, it will have a response. Any error will hit the `.catch` block and dispatch an upload error. since the catch block doesn't have a return value, `response` will be `null`

In either case we still want to clear the success or error message eventually

Reply:
I think linear try-catch-finally with async/await would be a bit easier to comprehend compared to the mix of async and promise catch. Wdyt?